File: cakk-api/src/main/java/com/cakk/api/dto/request/user/CertificationRequest.java
package com.cakk.api.dto.request.user;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import com.cakk.domain.mysql.dto.param.user.CertificationParam;
import com.cakk.domain.mysql.entity.user.User;

public record CertificationRequest(
	@NotBlank
	String businessRegistrationImageUrl,
	@NotBlank
	String idCardImageUrl,
	@NotNull
	Long cakeShopId,
	@NotBlank
	String emergencyContact,
	String message
) {

	public CertificationParam from(User user) {
		return CertificationParam.builder()
			.businessRegistrationImageUrl(businessRegistrationImageUrl)
			.idCardImageUrl(idCardImageUrl)
			.cakeShopId(cakeShopId)
			.emergencyContact(emergencyContact)
			.message(message)
			.user(user)
			.build();
	}
}

File: cakk-api/src/main/kotlin/com/cakk/api/dto/request/user/CertificationRequest.kt
package com.cakk.api.dto.request.user

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class CertificationRequest(
	@field:NotBlank
    val businessRegistrationImageUrl: String?,
	@field:NotBlank
    val idCardImageUrl: String?,
	@field:NotNull
    val cakeShopId: Long?,
	@field:NotBlank
    val emergencyContact: String?,
    val message: String?
)
